go/vt/vtgate/vindexes/vschema.go

```diff
@@ -461,23 +461,27 @@ func (vschema *VSchema) AddUDF(ksname, udfName string) error {
 	return nil
 }
 
-func buildGlobalTables(source *vschemapb.SrvVSchema, vschema *VSchema) {
+func BuildGlobalTables(source *vschemapb.SrvVSchema, vschema *VSchema, skipIfAlreadyGlobal bool) {
 	for ksname, ks := range source.Keyspaces {
 		ksvschema := vschema.Keyspaces[ksname]
 		// If the keyspace requires explicit routing, don't include any of
 		// its tables in global tables.
 		if ks.RequireExplicitRouting {
 			continue
 		}
-		buildKeyspaceGlobalTables(vschema, ksvschema)
+		buildKeyspaceGlobalTables(vschema, ksvschema, skipIfAlreadyGlobal)
 	}
 }
 
-func buildKeyspaceGlobalTables(vschema *VSchema, ksvschema *KeyspaceSchema) {
+func buildKeyspaceGlobalTables(vschema *VSchema, ksvschema *KeyspaceSchema, skipIfAlreadyGlobal bool) {
 	for tname, t := range ksvschema.Tables {
 		if gt, ok := vschema.globalTables[tname]; ok {
 			// There is already an entry table stored in global tables
 			// with this name.
+			if !skipIfAlreadyGlobal {
+				// Called when updating from schema tracking
+				continue
+			}
```

Comment:
The logic is fine, but doesn't the name of the variable seem exactly opposite of what it is doing? Like we are skipping adding to the global table if is already present when `skipIfAlreadyGlobal` is false? Am I misunderstanding this?

Reply:
You are right! I changed the name of the variable from `override` to this, which changed its meaning. So I should reverse the values passed in. Will do that.

Reply:
Fixed

Reply:
```suggestion
			if skipAlreadyGlobal {
				// Called when updating from schema tracking
				continue
			}
```

Reply:
`!` has to go I believe

Reply:
Duh, yes, thanks, needed to change this as well since I switched the logic around. Fixed and tests are now passing.